components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts

```diff
@@ -23,7 +23,12 @@ type LogEventTypes = {
  * @param taskId ID of the task to watch.
  */
 export function usePrebuildLogsEmitter(prebuildId: string, taskId: string) {
-    const [emitter] = useState(new ReplayableEventEmitter<LogEventTypes>());
+    const emitter = useMemo(
+        () => new ReplayableEventEmitter<LogEventTypes>(),
+        // We would like to re-create the emitter when the prebuildId or taskId changes, so that logs of old tasks / prebuilds are not mixed with the new ones.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [prebuildId, taskId],
+    );
```

Comment:
This is also important, it fixes a bug where logs would be duplicated with copies from previous prebuild runs when re-running.